tests/lbs/main.c

```diff
@@ -67,6 +70,29 @@ callback_append(void * cookie, int failed, int status, uint64_t blkno)
 	return (0);
 }
 
+/* Callback for APPEND request for "the starting block # is incorrect". */
+static int
+callback_append_should_bad_start(void * cookie, int failed, int status,
+    uint64_t blkno)
+{
+
+	(void)cookie; /* UNUSED */
+	(void)blkno; /* UNUSED */
+
+	/* Parsed the response? */
+	append_failed = failed;
+
+	/* The status should be "the starting block # is incorrect". */
+	if (status != 1)
+		append_failed = 1;
```

Comment:
Am I reading this correctly that `append_failed` is nonzero if the request failed to fail correctly?

Reply:
... yes?

If status is 0, this APPEND request did not reply in the expected manner.
If status is above 1, this APPEND request did not reply in the expected manner.
If status is exactly 1, then don't modify `append_failed`.

... admittedly, I tend to have the `INTERFACES` file open in a separate xterm whenever I'm working on kivaloo.  Should I change the comment to reflect that an APPEND response with a status of 1 indicates that "the starting block \# is incorrect"?

Reply:
NB: I used the
```
if (status != 1)
```

construct for ` callback_get_should_not_exist()` as well.  I could expand that to:

```
if (status == 1) {
        /* This is the desired behaviour. */
} else
       append_failed = 1;
```

instead.  (with `append_failed` changed to `get_failed` as appropriate)

Reply:
Also: I'm treating a status of 2 or more as being reserved for future use.  At the moment, `INTERFACES` only specifies a meaning for status of 1.

... I doubt that we'd add other status messages, but it's not impossible.  Like, APPEND of 2 could be "storage would exceed quota if this append was applied in full", or something like that.  It's unlikely, but not impossible, so I wrote the tests to allow for that expansion.

Reply:
Right, my concern was that the name `append_failed` was a bit ambiguous in the context of a request which is expected to return an error.  Can we add `append_bad_response` for the case of `status != 1` and reserve `append_failed` for lower-level failures (e.g. the server dropping the connection because the request was invalid)?